libbeat/common/string.go

```diff
@@ -0,0 +1,21 @@
+package common
+
+import (
+	"reflect"
+	"unsafe"
+)
+
+type NetString []byte
+
+// implement encoding.TextMarshaller interface to treat []byte as raw string
+// by other encoders/serializers (e.g. JSON)
+
+func (n NetString) MarshalText() ([]byte, error) {
+	return n, nil
+}
+
+func (n NetString) UnsafeString() string {
+	bh := (*reflect.SliceHeader)(unsafe.Pointer(&n))
+	sh := reflect.StringHeader{Data: bh.Data, Len: bh.Len}
+	return *(*string)(unsafe.Pointer(&sh))
```

Comment:
Not keen about unsafe territory either. That's why introduced NetString over unsafe copying all []byte into string. Purpose of this method is to be used with map lookups (map keys can not be []byte, but must be string) in cases of compiler not optimizing []byte into string conversion. Optimized version in disassembly contains runtime.slicebytetostringtmp, unoptimized with copy will be runtime.slicebytetostring).